src/eclingo/solver/generator.py

```diff
@@ -55,30 +60,41 @@ def __call__(self) -> Iterator[Candidate]:
                 fact(SA).
 
             hold(KA) :- kp_hold(KA).
+
+            positive_extra_assumptions(A) :- epistemic_atom(k(A), KA), kp_hold(KA).
+            % negative_extra_assumptions(A) :- epistemic_atom(k(A), KA), kp_not_hold(KA).
+
+            #show positive_extra_assumptions/1.
+            #show negative_extra_assumptions/1.
             """
 
-        self.control.add("base", [], self.reified_program)
```

Comment:
If changed to just reified_program, we can remove the assignment on line 18.

Reply:
We can do that